analyzers/src/SonarAnalyzer.CSharp/Rules/PrivateFieldUsedAsLocalVariable.cs

```diff
@@ -84,7 +83,8 @@ public sealed class PrivateFieldUsedAsLocalVariable : SonarDiagnosticAnalyzer
                 },
                 SyntaxKind.ClassDeclaration,
                 SyntaxKind.StructDeclaration,
-                SyntaxKindEx.RecordClassDeclaration);
+                SyntaxKindEx.RecordClassDeclaration,
```

Comment:
Line 75-82 is weird - unneeded indirection and `ToList`. Could be simplified like this:
```
                    privateFields.Keys
                        .Where(collector.IsRemovableField)
                        .ForEach(x => c.ReportIssue(Diagnostic.Create(Rule, privateFields[x].GetLocation(), x.Name)));
```